system/jhtree/ctfile.hpp

```diff
@@ -162,7 +169,7 @@ class jhtree_decl CKeyHdr : public CWritableKeyNode
     {
         return (hdr.nodeKeyLength != -1);
     }
-    inline char getKeyType() { return hdr.ktype; }
+    inline unsigned getKeyType() { return hdr.keyFlags; }
```

Comment:
It doesn't because SwapBigEndian ensures it is filled in.  I could leave the old bits where they are and only store new bits in the new flags.  e.g. always OR them together in this function.